src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -11418,90 +11418,19 @@ instruct testL_reg_imm(flagsRegCR0 cr0, iRegLsrc src1, uimmL16 src2, immL_0 zero
   ins_pipe(pipe_class_compare);
 %}
 
-instruct cmovI_conIvalueMinus1_conIvalue1(iRegIdst dst, flagsRegSrc crx) %{
-  // no match-rule, false predicate
-  effect(DEF dst, USE crx);
-  predicate(false);
+// Manifest a CmpL3 result in an integer register.
+instruct cmpL3_reg_reg(iRegIdst dst, iRegLsrc src1, iRegLsrc src2) %{
```

Comment:
I had forgotten one detail in my previous review. Sorry for that. We need to model the CR0 effect:
instruct cmpL3_reg_reg(iRegIdst dst, iRegLsrc src1, iRegLsrc src2, flagsRegCR0 cr0) %{
  match(Set dst (CmpL3 src1 src2));
  effect(KILL cr0);
(Same for other nodes.)

Reply:
Wow, thanks for catching it. But, let me make my naive question: why is it necessary?

Reply:
E.g. instruct testI_reg_imm sets cr0 as result and branchConFar uses it. The kill cr0 effect disallows scheduling your nodes between them.

Reply:
Btw. advantage of the expand rules is that the single instructions in a basic block can get scheduled in an interleaved fashion to hide instruction latencies. But this has become less relevant with modern out-of-order CPUs. We had done that for Power 6, but we're no longer optimizing for it.